app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -1,76 +1,359 @@
+                    messageText
+                )
+            )
+            val builder = DialogBuilderCreator("Change Password", layout)
+
+            val dialog = builder.create()
+            dialog.show()
+
+            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(View.OnClickListener {
+
+                var closeDialog = false
+
+                var passwordQualityCheck: String? =
+                    UtilityFunctions.validatePassword(newPasswordInput.text.toString())
+                var passwordEqualityCheck: String? = UtilityFunctions.confirmPassword(
+                    newPasswordInput.text.toString(),
+                    newPasswordConfirmInput.text.toString()
+                )
+
+                if (passwordQualityCheck != null) {
+                    messageText.text = passwordQualityCheck
+                } else if (passwordEqualityCheck != null) {
+                    messageText.text = passwordEqualityCheck
+                } else if (currentPasswordInput.text == null) {
+                    messageText.text = "Please enter current password"
+                } else {
+                    userProfileViewModel.changeUserPassword(
+                        currentPasswordInput.text.toString(),
+                        newPasswordConfirmInput.text.toString()
+                    )
+                    closeDialog = true
+                }
+
+                if (closeDialog) dialog.dismiss()
+            })
+
+        }
+
+        val addPhoneNumberFunction = View.OnClickListener {
+            val countryCodeInput = EditText(this)
+            countryCodeInput.inputType = InputType.TYPE_CLASS_NUMBER
+            countryCodeInput.filters = arrayOf(InputFilter.LengthFilter(5))
+            countryCodeInput.hint = "Enter Country Code"
+
+            val phoneInput = EditText(this)
+            phoneInput.inputType = InputType.TYPE_CLASS_PHONE
+            phoneInput.filters = arrayOf(InputFilter.LengthFilter(10))
+            phoneInput.hint = "Enter 10-digit phone number"
+
+            val messageText = TextView(this)
+            messageText.setTextColor(Color.RED)
+
+            val layout = LinearLayoutCreator(arrayOf(countryCodeInput, phoneInput, messageText))
+            val builder = DialogBuilderCreator("Add Phone Number", layout)
+            builder.setView(layout)
+
+            val dialog = builder.create()
+            dialog.show()
+
+            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(View.OnClickListener {
+
+                var closeDialog = false
+
+                val countryCodeValidationCheck =
+                    UtilityFunctions.validateCountryCode(countryCodeInput.text.toString())
+                val phoneNumberValidationCheck =
+                    UtilityFunctions.validatePhoneNumber(phoneInput.text.toString())
+
+                if (countryCodeValidationCheck != null) {
+                    messageText.setText(countryCodeValidationCheck)
```

Comment:
Use property access syntax like this:
![image](https://user-images.githubusercontent.com/64218770/211125719-4b8166eb-a00b-429c-909b-790fc4907aca.png)

You should check your code to use Kotlin semantics